commit 1189fd22c16be184a926b79eb35ad1d940183637
Author: Robert Mustacchi <rm@joyent.com>
Date:   Sat Feb 9 15:11:34 2013 -0800

    illumos support, but also define a solaris2.11 target to make gcc happier.

diff --git a/bfd/config.bfd b/bfd/config.bfd
index 3b9872a..0a8a383 100644
--- a/bfd/config.bfd
+++ b/bfd/config.bfd
@@ -280,7 +280,7 @@ case "${targ}" in
     targ_selvecs=bfd_elf32_bigarm_vec
     ;;
   arm-*-elf | arm-*-freebsd* | arm*-*-linux-* | arm*-*-conix* | \
-  arm*-*-uclinux* | arm-*-kfreebsd*-gnu | \
+  arm*-*-uclinux* | arm-*-kfreebsd*-gnu | arm-*-illumos | arm-*-solaris* | \
   arm*-*-eabi* )
     targ_defvec=bfd_elf32_littlearm_vec
     targ_selvecs=bfd_elf32_bigarm_vec
diff --git a/config.sub b/config.sub
index f9fcdc8..5bcdba9 100755
--- a/config.sub
+++ b/config.sub
@@ -1320,7 +1320,7 @@ case $os in
 	-gnu* | -bsd* | -mach* | -minix* | -genix* | -ultrix* | -irix* \
 	      | -*vms* | -sco* | -esix* | -isc* | -aix* | -cnk* | -sunos | -sunos[34]*\
 	      | -hpux* | -unos* | -osf* | -luna* | -dgux* | -auroraux* | -solaris* \
-	      | -sym* | -kopensolaris* \
+	      | -sym* | -kopensolaris* | -illumos* \
 	      | -amigaos* | -amigados* | -msdos* | -newsos* | -unicos* | -aof* \
 	      | -aos* | -aros* \
 	      | -nindy* | -vxsim* | -vxworks* | -ebmon* | -hms* | -mvs* \
diff --git a/gas/config/te-armillumoseabi.h b/gas/config/te-armillumoseabi.h
new file mode 100644
index 0000000..47b7698
--- /dev/null
+++ b/gas/config/te-armillumoseabi.h
@@ -0,0 +1,24 @@
+/* Copyright (c) 2013, Joyent, Inc.
+
+   This file is part of GAS, the GNU Assembler.
+
+   GAS is free software; you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as
+   published by the Free Software Foundation; either version 3,
+   or (at your option) any later version.
+
+   GAS is distributed in the hope that it will be useful, but
+   WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See
+   the GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with GAS; see the file COPYING.  If not, write to the Free
+   Software Foundation, 51 Franklin Street - Fifth Floor, Boston, MA
+   02110-1301, USA.  */
+
+#include "te-solaris.h"
+
+/* The EABI requires the use of VFP.  */
+#define FPU_DEFAULT FPU_ARCH_VFP
+#define EABI_DEFAULT EF_ARM_EABI_VER5
diff --git a/gas/configure.tgt b/gas/configure.tgt
index a171a32..7112eab 100644
--- a/gas/configure.tgt
+++ b/gas/configure.tgt
@@ -108,6 +108,8 @@ case ${generic_target} in
   arm-*-symbianelf*)			fmt=elf em=symbian ;;
   arm-*-kaos*)				fmt=elf ;;
   arm-*-conix*)				fmt=elf ;;
+  arm-*-*illumos*)			fmt=elf em=armillumoseabi ;;
+  arm-*-*solaris*)			fmt=elf em=armillumoseabi ;;
   arm-*-linux*aout*)			fmt=aout em=linux ;;
   arm-*-linux-*eabi*)			fmt=elf  em=armlinuxeabi ;;
   arm-*-linux-*)			fmt=elf  em=linux ;;
